packages/utils/addon/decorator.js

```diff
@@ -61,11 +61,16 @@ function convertStage1ToStage2(desc) {
     let kind = kindForDesc(desc);
     let placement = placementForKind(kind);
 
+    let { initializer } = descriptor;
+    delete descriptor.initializer;
+
     return {
       descriptor,
       key,
       kind,
       placement,
+      initializer,
```

Comment:
Should this field be conditionally set, so that `'initializer' in memberDescriptor === false`, if the original `descriptor` has no `initializer` in the first place?

Reply:
I think we could get away with `typeof` checks, better to be precise anyways.

Reply:
My primary concern was that Babel might not check for type, but mere presence. I recalled that Babel disallows the `initializer` property for some descriptors.

- https://github.com/babel/babel/blob/2bb24f996f025dccd26e1e5269d932efdeebc72d/packages/babel-helpers/src/helpers.js#L1629
- https://github.com/babel/babel/blob/2bb24f996f025dccd26e1e5269d932efdeebc72d/packages/babel-helpers/src/helpers.js#L1698

But looking at the implementation of [`_disallowProperty`](https://github.com/babel/babel/blob/2bb24f996f025dccd26e1e5269d932efdeebc72d/packages/babel-helpers/src/helpers.js#L1707-L1711) we're in the clear 😌

```js
function _disallowProperty(obj, name, objectType) {
  if (obj[name] !== undefined) {
    throw new TypeError(objectType + " can't have a ." + name + " property.");
  }
}
```